fastparquet/api.py

```diff
@@ -887,6 +888,10 @@ def categories(self):
             return self._categories
         if self.has_pandas_metadata:
             metadata = self.pandas_metadata
+            if "column_indexes" in metadata and len(metadata["column_indexes"]) > 0:
+                self._columns_dtype = metadata["column_indexes"][0]["numpy_type"]
```

Comment:
To be sure memory fails): does fastparquet write this metadata? Does arrow?

Reply:
Yes both do